lib/services/watcher.go

```diff
@@ -427,10 +461,10 @@ func (p *proxyCollector) broadcastUpdate(ctx context.Context) {
 
 // isInitialized is used to check that the cache has done its initial
 // sync
-func (p *proxyCollector) isInitialized() bool {
+func (p *proxyCollector) initializationChan() <-chan struct{} {
 	p.rw.RUnlock()
 	defer p.rw.RUnlock()
-	return p.initialized
+	return p.initializationC
 }
```

Comment:
Since we replaced the boolean field with a channel, why do we still need to return it under lock and these getter methods?